pkg/sql/plan2/explain/types.go

```diff
@@ -0,0 +1,173 @@
+// Copyright 2021 Matrix Origin
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package explain
+
+import (
+	"fmt"
+	"github.com/matrixorigin/matrixone/pkg/pb/plan"
+	"strings"
+)
+
+//type TableDef plan.TableDef
+//type ObjectRef plan.ObjectRef
+type Cost plan.Cost
+type Const plan.Const
+
+//type Expr plan.Expr
+
+//type Node plan.Node
+type RowsetData plan.RowsetData
+
+//type Query plan.Query
+
+type ExplainQuery interface {
+	ExplainPlan(buffer *ExplainDataBuffer, options *ExplainOptions)
+	ExplainAnalyze(buffer *ExplainDataBuffer, options *ExplainOptions)
+}
+
+//-------------------------------------------------------------------------------------------------------
+
+type NodeDescribe interface {
+	GetNodeBasicInfo(options *ExplainOptions) string
+	GetExtraInfo(options *ExplainOptions) []string
+	GetProjectListInfo(options *ExplainOptions) string
+	GetJoinConditionInfo(options *ExplainOptions) string
+	GetWhereConditionInfo(options *ExplainOptions) string
+	GetOrderByInfo(options *ExplainOptions) string
+	GetGroupByInfo(options *ExplainOptions) string
+}
+
+type FormatSettings struct {
+	buffer      *ExplainDataBuffer
+	offset      int
+	indent      int
+	indent_char byte
+	level       int
+}
+
+func NewFormatSettings() *FormatSettings {
+	return &FormatSettings{
+		buffer:      &ExplainDataBuffer{},
+		offset:      0,
+		indent:      2,
+		indent_char: ' ',
```

Comment:
suggest rename to indentChar

Reply:
ok, I will use the camel-case style to rename it